src/jdk.jpackage/windows/classes/jdk/jpackage/internal/WixAppImageFragmentBuilder.java

```diff
@@ -488,8 +488,7 @@ private String addShortcutComponent(XMLStreamWriter xml, Path launcherPath,
             xml.writeAttribute("Name", launcherBasename);
             xml.writeAttribute("WorkingDirectory", INSTALLDIR.toString());
             xml.writeAttribute("Advertise", "no");
-            xml.writeAttribute("Target", String.format("[#%s]",
-                    Component.File.idOf(launcherPath)));
+            xml.writeAttribute("Target", "[ARPMAINLAUNCHERLOCATION]");
```

Comment:
I don't think you can replace `launcherPath` with `[ARPMAINLAUNCHERLOCATION]`. `launcherPath` can reference any application launcher, i.e. it can be the main or one of the additional app launchers.